lib/mentat.ex

```diff
@@ -278,6 +270,17 @@ defmodule Mentat do
     Supervisor.stop(name)
   end
 
+  defp lookup(cache, key) do
```

Comment:
Initially, I copy/pasted the implementation from `get/2` into `fetch/4` and tweaked it like this, because it didn't have the information it needed just by calling `get` like it was doing before. Then, I extracted the result here because I thought it worked well for both and also simplified the implementation of `get/2`